File: src/main/java/phss/wsworkcars/repositories/ModeloRepository.java
package phss.wsworkcars.repositories;

import org.springframework.data.jpa.repository.JpaRepository;
import phss.wsworkcars.models.Modelo;

import java.util.List;
import java.util.Optional;

public interface ModeloRepository extends JpaRepository<Modelo, Long> {
    Optional<Modelo> findByNome(String nome);
    List<Modelo> findByMarcaId(long marcaId);
}

File: src/main/kotlin/phss/wsworkcars/repositories/ModeloRepository.kt
package phss.wsworkcars.repositories

import org.springframework.data.jpa.repository.JpaRepository
import phss.wsworkcars.models.Modelo

interface ModeloRepository : JpaRepository<Modelo, Long> {
    fun findByNome(nome: String): Modelo?
    fun findByMarcaId(marcaId: Long): List<Modelo>
}